### ISSUE TITLE
MeshCtrl.JS file upload sticks on larger files

### ISSUE THREAD

**Issue by adsgreen** [Ylianst/MeshCentral#2142]
Just looking at using the mshctrl.js "upload" command to populate some files on the remote endpoints we have and noticed some odd behaviour.

For reference I'm issuing the command:
```
node meshctrl.js upload --url wss://xxxxx.com --id "fneK@..........." --file "c:\\Temp\\aFile.zip" --loginuser xxxxx  --loginpass xxxxx --target "c:\\Backup\\Test"
Connecting...
Waiting for Agent...
Connected.
```
If the file "aFile.zip" is relatively small (around a 1mb ish) then the upload works perfectly.
However on larger files it seems to stall.... and just be stuck here. I can see the file on the remote machine has stopped uploading. So some data has arrived but it's now out of sync.

Interestingly it will always fail at the same place for a given file but this place isn't the same for all files.

Adding some logging to meshctrl.js upload section:
Connecting...
Waiting for Agent...
Connected.
Uploading... 4096
Uploading... 8192
Uploading... 12288
Uploading... 16384
Uploading... 20480
Uploading... 24576
Uploading... 28672
Uploading... 32768
Uploading... 36864
Uploading... 40960
Uploading... 45056
Uploading... 49152
Uploading... 53248
Uploading... 57344
Uploading... 61440

And it stops here and never recovers.
Retrying the upload fails in the same place, everytime on all agents.

I can improve things slightly by changing the meshctl.js file:
```
if (cmd.reqid == 'up') {
                    if ((cmd.action == 'uploadack') || (cmd.action == 'uploadstart')) {
                        var buf = Buffer.alloc(4096);
                        var len = require('fs').readSync(settings.uploadFile, buf, 0, 4096, settings.uploadPtr);
```
If I change the buffer size from 4096 to something higher (much higher) then the files seem to arrive ok but if it's not high enough it will fail in a different part of the upload.

Using a variety of numbers for the buffer size, it appears to get stuck in different places. It's almost as if the data that is being read is causing a problem. Note: dividing the buffer by 2 will always cause the same problem either at the same place as the larger buffer or sooner.
I've looked in at a hex editor and can't see anything obvious with the source file (in this case its MeshCentralInstaller-2.9.exe).

**Comment by Ylianst** [comment]
Interesting, I will take a look at that.

**Comment by Ylianst** [comment]
Thanks for reporting this. Just published MeshCentral v0.7.39 with a fix for this. Let me know if it works.

**Comment by adsgreen** [comment]
Looks good - I was just looking at the binary prefix in the agent thinking "aha" when you fixed it :)
